### ISSUE TITLE
Component: Events

### ISSUE THREAD

**Issue by lloan** [inland-empire-software-development/main#21]
This is a complex component. It has a couple of moving parts that make it complex. 

There are 3 parts to this component:
- series events (omit for v1)
- single events (omit presented by text)
- event timeline 

Data you need from Meetup.com for each event:
- date
- time
- title 
- link to event 

This will require that you check Meetup to see if we have a series going on. However, for v1, we might just omit this section and focus on the single events. Retrieve all events for the current month and render the 4 upcoming events. The timeline will require that you render a circle for each event and their placement depends on the day of the event. In the image below, May is at the top and April is at the bottom. An event that is on April 16 will be about halfway in between those. An event that happens on April 28th would be much closer to the May text. When you hover on any of these, it should have a hover state (fills with gradient color) and tool tip has the date for the event. If user clicks on it, the 4 rendered events should lead with the event clicked and the 3 events right behind that one. If there are only 2 events, render only 2 events. 

![image](https://user-images.githubusercontent.com/3924990/59503951-8a4acb80-8e56-11e9-93f4-7a1d9f3d4c79.png)

**Comment by tonynguyen111997** [comment]
Link to the meetup api

**Comment by Holyoke** [comment]
I've investigated the authorization flow, and there are caveats. The meetup api does return results, which you can test on the console. They're stored in a results array, and we can derive event data fine to populate the components - the get request would be like 

```
https://api.meetup.com/2/events&group_urlname=iesd-meetup
```
and the response looks like this
![image](https://user-images.githubusercontent.com/6300995/59818050-27828580-92d7-11e9-8cb5-a902176c11f1.png)



Here's the authorization process for Meetup's API : https://www.meetup.com/meetup_api/auth/

The issue is that the main authentication process for client-side apps requires a redirect/implicit flow to obtain authorization, which requires the app to redirect the user to sign-in with their meetup logins, which is not ideal in my opinion. The ideal solution would be the Server Flow, where an API key is obtained via a normal request. 

However, that feature is only available to Meetup pro users. All of this is to extract data leaning on meetups services, but an alternative is to locally store information instead. The trade-off is that administrators will have to manually sync the ie-sd website events page. 

So in terms of getting data from meetup, there are authorization caveats. Since authorization seems to be a bit involved, perhaps the issue can be split-up into presentational components and authorization.

**Comment by dsacosta** [comment]
@Holyoke Hey Jason, please update me on this issue and let me know if you need any help.

**Comment by Holyoke** [comment]
@dsacosta hey David, I'm currently blocked on CSS. I have setup the React code, so everything is working. The only thing left is styling it correctly. Here's how it look at the moment

![image](https://user-images.githubusercontent.com/6300995/61584505-b1ec1e00-aafd-11e9-9665-93ea0fa11142.png)

My work can be seen here at this branch
https://github.com/Holyoke/main/tree/events-jason

**Comment by dsacosta** [comment]
@VelazquezLuis Can you help Jason with this?

**Comment by VelazquezLuis** [comment]
spoke with him today, @Holyoke , if you need any more help let me know.

**Comment by lloan** [comment]
Hey @Holyoke - how's it going with this component, any blockers?

**Comment by Holyoke** [comment]
Sorry, I haven't been working with on the component lately, but this weekend I will. 

last it stands the swiper version is being worked on - I have a basic way of implementing it based off Luis and Robin's work on their components, so I will be basing the mobile version off that. 

I hope to get the mobile version done soon, but the normal desktop version should be working with this branch, and that can be reviewed.

**Comment by lloan** [comment]
Closing because this is no longer needed.
